File: leetcode/easy/ValidAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] counts = new int[26];
        for (int i = 0; i < s.length(); i++) {
            int idx = (int)(s.charAt(i) - 'a');
            counts[idx]++;
        }
        for (int i = 0; i < t.length(); i++) {
            int idx = (int)(t.charAt(i) - 'a');
            counts[idx]--;
        }
        for (int count : counts) {
            if (count != 0) {
                return false;
            }
        }
        return true;
    }
}

File: leetcode/easy/ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        counts = [0] * 26
        for c in s:
            counts[ord(c) - 97] += 1  # ord('a'): 97
        for c in t:
            counts[ord(c) - 97] -= 1  # ord('a'): 97
        for c in counts:
            if c != 0:
                return False
        return True
